stellargraph/core/graph_networkx.py

```diff
@@ -775,6 +775,36 @@ def to_adjacency_matrix(self, nodes: Optional[Iterable] = None):
             self._graph, dtype="float32", weight=self._edge_weight_label, format="coo"
         )
 
+    def to_networkx(self):
+        # Despite this class using NetworkX, this implementation does not directly use that
+        # representation, so that it can be reused as we move away from being NetworkX-based.
+        if self.is_directed():
+            graph = nx.MultiDiGraph()
+        else:
+            graph = nx.MultiGraph()
+
+        types = self.node_types
+
+        for ty in types:
+            node_ids = self.nodes_of_type(ty)
+            ty_dict = {self._node_type_attr: ty}
+
+            if ty in self._node_attribute_arrays:
+                # has features!
+                features = self.node_features(node_ids, node_type=ty)
+
+                for node_id, node_features in zip(node_ids, features):
+                    graph.add_node(
+                        node_id, **ty_dict, **{self._feature_attr: node_features},
+                    )
+            else:
+                # no features, so just add the type
+                graph.add_nodes_from(node_ids, **ty_dict)
+
+        graph.add_edges_from(self.edges(triple=True))
```

Comment:
Since `self.edges(triple=True)` returns node pairs with `edge_type` and not the actual weight/value over the edge, `graph.add_edges_from(self.edges(triple=True))` results in a networkx multigraph with values that are `edge_type` and not the actual values over the edges. I have tested it only for one `edge_type`, not sure what happens in case of multi-edge type graph. 
Moreover, as @huonw you mentioned in #642 the `edge_type` is not the actual label of the type of edge.

Reply:
Ah, yeah; this is broken for two reasons:

1. `.edges(triple=True)` is wrong, but that's being fixed separately in #642 #643 
2. even when it returns the right tuples, the edge type will be passed in as a key, not a data dictionary